### ISSUE TITLE
pandas import errors with current bottleneck pip wheel

### ISSUE THREAD

**Issue by stas00** [pydata/bottleneck#204]
```
conda create --name bottleneck python=3.7
conda activate bottleneck
pip install pandas bottleneck
Successfully installed bottleneck-1.2.1 numpy-1.15.4 pandas-0.23.4 python-dateutil-2.7.5 pytz-2018.9 six-1.12.0
```
then:
```
python -c "import pandas"
```
gives:
```
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'
```
In the example, the error is not fatal, in normal settings, but in some settings it leads to a fatal error.

Different workarounds:

1. `conda install bottleneck numpy` (instead of pip)
2. pip install bottleneck from source 
```
pip install git+https://github.com/kwgoodman/bottleneck/commit/104778a8dea49d0ca230288b5011c17979c4ac99
```
3. `pip install numpy==1.16.0rc1`

There are quite a few of these reports on [google](https://www.google.com/search?q=ModuleNotFoundError%3A+No+module+named+%22numpy.core._multiarray_umath%22)

**Comment by stas00** [comment]
Here is an important insight, as I suspected in https://github.com/kwgoodman/bottleneck/issues/191#issuecomment-453773941 it was about a binary, despite pypi having only the source. The problem happens if the user has ever `pip install bottleneck` on their system, pip will cache the binary and on the next  `pip install bottleneck`  it'll use that instead of building a new one. And if numpy has since changed, pip has no way of invalidating that cached binary wheel.

So for example here is workaround #4:
```
pip install bottleneck --no-binary :all:
# or with the same effect:
pip install bottleneck --no-cache-dir
```
and now `python -c "import pandas"` gives no error. But it's still a workaround and not a proper solution.

**Comment by qwhelan** [comment]
@stas00 Thanks for documenting this - I've included your workaround 4 as the default error message if users encounter this going forward.

I'd like to programmatically prevent this from occurring, but I'm guessing anyone on py2.7 probably is not on numpy 1.16. Once we drop 2.7, I'd be very comfortable setting a min numpy version of 1.16, which would work until they introduce another breaking change.

**Comment by qwhelan** [comment]
This should be resolved by the 1.3.0 release - please feel free to re-open if not the case.
